### DISCUSSION TITLE
Source directory does not exist

### DISCUSSION THREAD

**Discussion by houcros** [RogueOneEcho/caesura#194] [Support] [answered]
I've been using caesura frequently. Since yesterday, I'm getting these errors:

```
$ docker compose run --rm caesura verify <id>
2026-02-24 00:13:34.262 WARN △ Unsuitable for transcoding unknown
2026-02-24 00:13:34.262 WARN △ Source directory does not exist: Artist - Album (Year) FLAC

$ docker compose run --rm caesura transcode --verbosity trace <id>
2026-02-24 00:13:50.669 TRCE ↩ Sending request GET /ajax.php?action=torrent&id=<id>
2026-02-24 00:13:50.975 TRCE ↩ Received response after 0.306
2026-02-24 00:13:50.978 TRCE ↩ Received 200 OK response without error
2026-02-24 00:13:50.978 TRCE ↩ Sending request GET /ajax.php?action=torrentgroup&id=<id>
2026-02-24 00:13:51.215 TRCE ↩ Received response after 0.237
2026-02-24 00:13:51.217 TRCE ↩ Received 200 OK response without error
2026-02-24 00:13:51.222 ERRO ! Failed to get source from options
2026-02-24 00:13:51.222 ERRO ! Source directory does not exist: Artist - Album (Year) FLAC
```

I would swear I didn't change anything in my config or my overall setup lately. The only things that comes to mind: I enabled cross-seeding in Qui yesterday. I wonder if it might somehow be related. Otherwise, do you have any idea what this can be? It caught my eye that after "Source directory does not exist" it prints the directory and not the full path to it. Not sure if relevant.

More details:
```
$ docker compose run --rm caesura --version
caesura 0.26.0

$ cat docker-compose.yml
services:
  caesura:
    container_name: caesura
    image: ghcr.io/rogueoneecho/caesura
    volumes:
    #- ./config.yml:/config.yml:ro
    - /mnt/torrents/complete:/content
    - /mnt/torrents/watch:/watch
    - ./output:/output
    - ./cache:/cache
    configs:
    - source: caesura_config
      target: /config.yml
configs:
  caesura_config:
    content: |
      announce_url: [redacted]
      api_key: [redacted]
      content:
      - /content
      copy_torrent_to: /watch
      verbosity: debug
```

**Comment by RogueOneEcho** [comment]
Is this RED or OPS?

**Reply by RogueOneEcho** [reply]
Are you certain `/mnt/torrents/complete/Artist - Album (Year) Flac` exists?
Has qui done anything to that directory? Changed the permissions, replaced it with a symlink?
What user is the Docker container running as? Does it definitely have read access to that dir?

**Reply by houcros** [reply]
The directory definitely exists. I checked both on the mount and on a busybox Docker image with the same volume mount, to see it from the container's perspective. (`ls -laL '/mnt/torrents/complete/' | grep ...)` and `ls -laL /content | grep ...`)
As far as I can tell, qui didn't do anything with that directory: it's not a symlink, permissions are `drwxr-xr-x`.
```
drwxr-xr-x 2 1000 1000           0 Feb 23 15:26 Artist - Album (Year) Flac
```
Not sure which user Docker is running as, I presume root. whoami is root, and I never had the user in the docker-compose. To solve the issue, I  did also try adding the user to docker-compose:
```
caesura:
    container_name: caesura
    image: ghcr.io/rogueoneecho/caesura
    user: 1000:1000
```
but it changed nothing.

**Reply by RogueOneEcho** [reply, marked as answer]
Questions: 

1. Is this RED or OPS?
2. What is the ID number of the torrent you're having issues with?

Possibly solutions:

a) Most likely explanation is the API is returning a path that contains invisible characters. 

qBittorrent (via libtorrent) automatically strips these but caesura v0.26.0 doesn't (there's already a fix lined up for the next release)

See https://github.com/RogueOneEcho/caesura/issues/162#issuecomment-3434368589

You can check by copying the `caesura` console output into a code editor with a "show hidden/invisible chars functionality"

b) I know you're using placeholder values but worth noting that the API is saying `Artist - Album (Year) FLAC` but you local directory is saying `Artist - Album (Year) Flac` (`Flac` vs `FLAC`) that is another mismatch that could cause issues

**Reply by houcros** [reply]
Ok it seems it's a) as you suggested. I stored the output in a file and then with GNU cat (`-v, --show-nonprinting: use ^ and M- notation, except for LFD and TAB`):
```
$ cat -v out.txt
^[[90m2026-02-24 10:13:05.078 ^[[0m^[[31mERRO^[[0m ^[[31m!^[[0m ^[[1mFailed^[[0m to get source from options
^[[90m2026-02-24 10:13:05.078 ^[[0m^[[31mERRO^[[0m ^[[31m!^[[0m Source directory does not exist: Artist - Album (Year) FLAC (Label M-bM-^@M-^NM-bM-^@M-^S Catalog)
```
I also pasted it in [this site](https://invisiblecharacterviewer.com). It seems there's an n-dash at the end of this string: "(Label – Catalog)". (TIL the n-dash is a non-printable character.)

Also, this is partly user error. I did try another transcode, to see if it was problem with this release or in general. The other one failed for the same reason, but now I realise that one was my fault because I actually didn't have that file. That tripped me into thinking the whole caesura stopped working for me, but it seems it was just failing on this particuar one because of non-printable characters. I'm sorry for that red herring. :/

(BTW I'm not ignoring your first questions, just a bit paranoid about sharing too many details here. I could answer those questions, but now that I provided the name of the release, I'm hoping those answers are not very relevant any longer?)

Thank you for taking the time to help me out. IIUC there's nothing to be done but wait for an upcoming realease where this is handled, correct?

**Reply by RogueOneEcho** [reply]
> IIUC there's nothing to be done but wait for an upcoming realease where this is handled, correct?

Correct. 

I've checked and can confirm this works in the upcoming version:

```
docker compose run --rm caesura verify ID_NUMBER --verbosity trace
Container caesura-caesura-run-fa9827933430 Creating 
Container caesura-caesura-run-fa9827933430 Created 
2026-02-24 13:40:43.608 DBUG * Loaded 148 CA root certificates from the system
2026-02-24 13:40:43.661 TRCE ↩ Sending request GET /ajax.php?action=torrent&id=ID
2026-02-24 13:40:44.036 TRCE ↩ Received response after 0.375
2026-02-24 13:40:44.037 TRCE ↩ Received 200 OK response without error
2026-02-24 13:40:44.037 TRCE ↩ Sending request GET /ajax.php?action=torrentgroup&id=GROUP_ID
2026-02-24 13:40:44.262 TRCE ↩ Received response after 0.226
2026-02-24 13:40:44.264 TRCE ↩ Received 200 OK response without error
2026-02-24 13:40:44.266 DBUG * Verifying ALBUM NAME
2026-02-24 13:40:44.267 TRCE ↩ Found 5 flacs in: /path/to/content
```

Just so you're aware that all files in that release are faulty but that's unrelated to the issue you were seeing:

```
2026-02-24 13:40:45.565 WARN △ Unsuitable for transcoding ALBUM
2026-02-24 13:40:45.565 WARN △ FLAC stream error: Ill-formed FLAC stream: stream starts with ID3 header rather than FLAC header:  /path/to/file
```

---

You can now edit you message and remove the file name if you would prefer it be concealed.

1. Click ... in top right
2. Delete the file name
3.  Update comment
4. Click "edited" in top right
5. Select the first revision.
6. Scroll down and click "delete revision"

**Reply by houcros** [reply]
Thanks for checking that: good to know it will work in the next version. Bummer that the flacs are borked; guess I'll just forget about that release. :/ Also thanks for the tip on how to clean-up my message, didn't know that was possible. Done that now. Much obliged for your time and help. :)
